File: ruby/kata_spec.rb
require "rspec"
require_relative "kata"

describe CodeKata do
  describe "#test" do
    it "returns true" do
      expect(subject.test_method).to be true
    end
  end
end

describe "#test" do
  it "returns my_test string" do
    expect(test).to eq "my_test"
  end
end

File: javascript/kata.spec.js
'use strict';

const CodeKata = require('./kata').CodeKata;
const myTest = require('./kata').myTest;
const expect = require('chai').expect;


describe('CodeKata', function() {
    describe('#testMethod', function() {
        it('should return True', function() {
            const codeKata = new CodeKata;
            expect(codeKata.testMethod()).to.be.true;
        });
    });
});

describe('#test', function() {
    it('should return 3', function() {
        expect(myTest()).to.be.equal('my_test');
    });
});
